java-util/src/main/java/io/druid/java/util/common/parsers/TimestampParser.java

```diff
@@ -47,9 +49,17 @@
         public DateTime apply(String input)
         {
           Preconditions.checkArgument(input != null && !input.isEmpty(), "null timestamp");
+
           for (int i = 0; i < input.length(); i++) {
             if (input.charAt(i) < '0' || input.charAt(i) > '9') {
-              return parser.parseDateTime(ParserUtils.stripQuotes(input));
+              DateTimeZone dateTimeZone = DateTimeZone.UTC;
+              int lastIndex = input.lastIndexOf(" ");
+              if (lastIndex > 0 && ParserUtils.isValidTimeZone(input.substring(lastIndex + 1))) {
+                dateTimeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(input.substring(lastIndex + 1)));
+                input = input.substring(0, lastIndex);
+              }
+
+              return new DateTime(parser.parseDateTime(ParserUtils.stripQuotes(input)), dateTimeZone);
```

Comment:
If the string may appear to be quoted, `stipQuotes()` should go before determining last index of `' '` and taking substring, because we may end up trying to parse time zone `UTC"` or `UTC'`

Reply:
Please add a test about this